File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        left = 0
        right = len(s)-1

        while left<right :

            while left<right and not self.isalphanum(s[left]) :
                left += 1
            while left<right and not self.isalphanum(s[right]):
                right -= 1
            if s[left].lower() != s[right].lower() :
                return False
            left += 1
            right -= 1
        
        return True

    def isalphanum(self, ch)->bool :
        if ((ord('A')<=ord(ch)<=ord('Z')) or 
        (ord('a')<=ord(ch)<=ord('z')) or
        (ord('0')<=ord(ch)<=ord('9'))):
            return True

    

        

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {

       int l = 0;
       int r = s.size();

       while(l<r)
       {
            while(l<r && !isalnum(s[l])) //isalnum is built-in function
                l++;
            while(l<r && !isalnum(s[r]))
                r--;
            if(tolower(s[l]) != tolower(s[r])) //tolower is built-in function
            {
                return false;
            } 
            l++;
            r--;
       }
       return true; 
    }
};